builder-base/scripts/download_golang.sh

```diff
@@ -27,44 +27,63 @@ RELEASE_NUMBER="$(echo $VERSION | cut -d'-' -f 2)"
+      curl -sSLf --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/$arch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.$arch.rpm.sha256 -o $filename.sha256
 
-            if [[ $(sha256sum ${filename} | cut -d' ' -f1) != $(cut -d' ' -f1 "${filename}.sha256") ]] ; then 
-                echo "Checksum doesn't match!"
-                exit 1
-            fi
-        fi
-    done
+      if [[ $(sha256sum ${filename} | cut -d' ' -f1) != $(cut -d' ' -f1 "${filename}.sha256") ]]; then
+        echo "Checksum doesn't match!"
+        exit 1
+      fi
+    fi
+  done
 
-    for artifact in golang-docs golang-misc golang-tests golang-src; do
-        local filename="$outputDir/$arch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm"
-        if [ ! -f $filename ]; then
-            curl -sSLf --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/noarch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm -o $filename --create-dirs
-            curl -sSLf --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/noarch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm.sha256 -o $filename.sha256
+  for artifact in golang-docs golang-misc golang-tests golang-src; do
+    local filename="$outputDir/$arch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm"
+    if [ ! -f $filename ]; then
+      curl -sSLf --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/noarch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm -o $filename --create-dirs
+      curl -sSLf --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/noarch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm.sha256 -o $filename.sha256
 
-            if [[ $(sha256sum ${filename} | cut -d' ' -f1) != $(cut -d' ' -f1 "${filename}.sha256") ]] ; then 
-                echo "Checksum doesn't match!"
-                exit 1
-            fi
-        fi
-    done
+      if [[ $(sha256sum ${filename} | cut -d' ' -f1) != $(cut -d' ' -f1 "${filename}.sha256") ]]; then
+        echo "Checksum doesn't match!"
+        exit 1
+      fi
+    fi
+  done
 }
 
-if [[ $ARCHITECTURE =~ "linux/amd64" ]] ; then 
-    build::go::download "${VERSION}" "$OUTPUT_DIR" "x86_64"
-fi
+function build::go::download {
+  # Set up specific go version by using go get, additional versions apart from default can be installed by calling
+  # the function again with the specific parameter.
+  local version=${1%-*}
+  local outputDir=${2}
+  local archs=${3}
+
+  for arch in ${archs/,/ }; do
+    local filename="$outputDir/${arch}/go$version.${arch/\//-}.tar.gz"
+    if [ ! -f $filename ]; then
+      curl -sSLf --retry 5 "https://go.dev/dl/go$version.${arch/\//-}.tar.gz" -o $filename --create-dirs
```

Comment:
it would probably be best to check the checksums here. if you want to follow the pattern we use elsewhere, we could check those in or download them from go.dev, like we did the rpm shas above.  again but not strictly required but would be at least slightly better

Reply:
Made the additions for checking the sha and here is the local output.
```
+ curl -sSLf --retry 5 https://go.dev/dl/go1.22.4.linux-amd64.tar.gz -o /tmp/test/linux/amd64/go1.22.4.linux-amd64.tar.gz --create-dirs
++ curl -sSLf --retry 5 'https://go.dev/dl/?mode=json'
++ jq -r --arg tar go1.22.4.linux-amd64.tar.gz '.[].files[] | if .filename == $tar then .sha256 else "" end'
++ xargs
+ sha256sum=ba79d4526102575196273416239cca418a651e049c2b099f3159db85e7bade7d
++ sha256sum /tmp/test/linux/amd64/go1.22.4.linux-amd64.tar.gz
++ cut '-d ' -f1
+ [[ ba79d4526102575196273416239cca418a651e049c2b099f3159db85e7bade7d != \b\a\7\9\d\4\5\2\6\1\0\2\5\7\5\1\9\6\2\7\3\4\1\6\2\3\9\c\c\a\4\1\8\a\6\5\1\e\0\4\9\c\2\b\0\9\9\f\3\1\5\9\d\b\8\5\e\7\b\a\d\e\7\d ]]
```